components/eam/src/physics/cam/micro_p3.F90

```diff
@@ -2936,7 +2936,7 @@ subroutine nr_conservation(nr,ni2nr_melt_tend,nr_ice_shed_tend,ncshdc,nc2nr_auto
   !recall that melting number is scaled by nmltratio to account for ice crystals melting then
   !rapidly evaporating. Thus ni removal from melting is *not* scaled by nmltratio...
   source_nr = nr + (ni2nr_melt_tend*nmltratio + nr_ice_shed_tend + ncshdc + nc2nr_autoconv_tend)*dt
-  if(sink_nr > source_nr) then
+  if(sink_nr > source_nr .and. sink_nr.gt.0._rtype) then
```

Comment:
I think these should probably be != 0.

Reply:
Does the original code actually cause an FPE? I think source_nr must be >= 0. Thus, if sink_nr > source_nr, then sink_nr > 0, causing no issues in source_nr/sink_nr.

Reply:
@ambrad , yes. My methodology was:
1) Build SCREAM with SCREAM_FPE
2) RUN ctest
3) Pick one of the failing tests
4) Run it through gdb, find FPE line, fix it, run it again, repeat until test passes

So, I don't think I touched anything in this PR unless it was to fix an FPE. I can double-check this particular change.

Reply:
@ambrad , I stand corrected. This check was not necessary.

Reply:
@ambrad sorry for the confusion; this change does appear to be necessary (I tested my non-FPE build on accident when I said it was not necessary).

Reply:
Is source_nr going negative? If it is, we should bring this to the attention of Aaron. Otherwise, I'm not seeing how this code triggers an FPE.

Reply:
@ambrad , I remove the `.and. sink_nr.gt.0._rtype` and added a print inside the if block. The values printed immediately before the FPE crash were:

```
sink_nr= -0.00000000000000000000000000000000E+00 source_nr= -3.63189856026829738233629996102536E-01
```

Reply:
@AaronDonahue is source_nr < 0 and sink_nr = 0 a physically valid state? If not, we need to figure out what's leading to it.